src/playground/public/assets/js/app.ts

```diff
@@ -29,12 +29,18 @@ async function printAskVersion(elemId: string, askScriptServerUrl: string) {
 function registerAskScriptEditor(
   editorElementId: string,
   runElementId: string,
-  askScriptServerUrl: string
+  askScriptServerUrl: string,
+  dirty: boolean
```

Comment:
Have you thought of a use-case in which `dirty` will be `true` when you call this function?

Reply:
@YonatanKra no, I think that function should be initialized only once with default `false dirty`  when we start the playground. Should add some more? some conditions?

Reply:
Yea, move it inside. Much cleaner.
Also, remove the `change` listener once it is done.
Something like that:
```
const changeListenerCB = () => {
  confirmExitFromPlayground();
  editor.session.off('change', changeListenerCB);
}
editor.session.on('change', changeListenerCB);
```
Then you don't need the dirty flag at all. You just set the confirmation once a change happens and remove the change listener when you are done.
I'm not keen on the editor's API, so the syntax might vary.

Reply:
@YonatanKra good point.
have some problem with `off`
Error code Property 'off' does not exist on type 'IEditSession'.
So I try to understand these problems in I found something like this https://github.com/thlorenz/brace/blob/master/index.d.ts#L458 . I can add `@ts-ignoge for this line or use `editor.off`
FYI: The `session.removeListener` throws the same error.

Maybe you have more experience with the ace editor.

Reply:
Hi. Sorry. Didn't get the time to look at the editor api. Never used it before.
If you can't find a way to do it using the wditor api I'll take a look during the weekend.